cfme/infrastructure/host.py

```diff
@@ -452,7 +448,12 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+        from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+        dest = 'Hosts'
+        if hasattr(self.obj, 'provider') and self.obj.provider:
```

Comment:
I don't think you need a `hasattr` call here, the parameter is optional but the class attribute will be set regardless of if its None or not.

Looks like `if self.obj.provider` is sufficient, since it will either be None or a provider object.